### PR TITLE
Feature/auto enable e2ee for all participants

### PR DESCRIPTION
**Pull request by tmoldovan8x8**

### CODE REVIEW

File: react/features/e2ee/middleware.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
     case TOGGLE_E2EE: {
         const conference = getCurrentConference(getState);
 
-        if (conference) {
+        if (conference && conference.isE2EEEnabled() !== action.enabled) {
```
Comment (saghul) [inline react/features/e2ee/middleware.js:41]: Is this really needed? It is already checked internally...

Comment (tmoldovan8x8) [reply react/features/e2ee/middleware.js:41]: We also dispatch the sounds actions here, plus some other stuff, which we don't want unless enabled has changed.

Comment (saghul) [reply react/features/e2ee/middleware.js:41]: Makes sense!

File: react/features/base/participants/middleware.js [hunk 2]
```diff
@@ -270,17 +271,20 @@ StateListenerRegistry.register(
...
     const e2eeEnabled = newValue === 'true';
 
-    dispatch(participantUpdated({
-        conference,
-        id: participantId,
-        e2eeEnabled
-    }));
+    if (e2eeEnabled) {
+        dispatch(toggleE2EE(e2eeEnabled));
+        dispatch(participantUpdated({
```
Comment (saghul) [inline react/features/base/participants/middleware.js:282]: We need to do this also for remote participants, please take it out of the if.

**Review by saghul — APPROVED** [review]
🚀

**Pull request merged**